ui/plugin-system/src/components/TimeSeriesQueryEditor.tsx

```diff
@@ -11,24 +11,137 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import { Box, BoxProps } from '@mui/material';
-import { TimeSeriesQueryDefinition } from '@perses-dev/core';
+import { useState } from 'react';
 import { produce } from 'immer';
+import { Box, BoxProps, Button, IconButton, Stack, Typography } from '@mui/material';
+import AddIcon from 'mdi-material-ui/Plus';
+import DeleteIcon from 'mdi-material-ui/DeleteOutline';
+import ChevronDown from 'mdi-material-ui/ChevronDown';
+import ChevronRight from 'mdi-material-ui/ChevronRight';
+import { TimeSeriesQueryDefinition, QueryDefinition } from '@perses-dev/core';
+import { usePlugin, usePluginRegistry } from '../runtime';
 import { PluginEditor, PluginEditorProps } from './PluginEditor';
 
+const DEFAULT_QUERY_PLUGIN_TYPE = 'TimeSeriesQuery';
+
+export interface TimeSeriesQueryEditorProps {
+  queries: TimeSeriesQueryDefinition[];
+  onChange: (queries: QueryDefinition[]) => void;
+}
+
+export function TimeSeriesQueryEditor(props: TimeSeriesQueryEditorProps) {
```

Comment:
This used to be `TimeSeriesChartQueryEditor`

Reply:
I think this is ok for now, but eventually this will probably just change to `QueryEditor`. Once we have multiple types, you may want to have one query with logs and one with metrics so it should support more than `TimeSeriesQuery`